elfpy/time/time.py

```diff
@@ -154,34 +154,97 @@ def time_to_days_remaining(time_remaining: float, time_stretch: float = 1, norma
     return normed_days_remaining * normalizing_constant
 
 
+TimeUnit = Literal["seconds", "minutes", "hours", "days", "years"]
+
+
 @dataclass
 class BlockTimeFP:
-    r"""Global time."""
+    r"""State class for tracking block timestamps and global time"""
 
-    time: FixedPoint = FixedPoint(0)  # time in years
-    block_number: FixedPoint = FixedPoint(0)
-    step_size: FixedPoint = FixedPoint("1.0") / FixedPoint("365.0")  # defaults to 1 day
+    _time: FixedPoint = FixedPoint(0)
+    _block_number: FixedPoint = FixedPoint(0)
+    _step_size: FixedPoint = FixedPoint("1.0") / FixedPoint("365.0")  # defaults to 1 day
+    time_unit: str = "years"
 
-    @property
-    def time_in_seconds(self) -> FixedPoint:
-        """1 year = 31,556,952 seconds"""
-        return self.time * FixedPoint("31_556_952.0")
+    def __post_init__(self):
+        if self.time_unit != "years":
+            raise ValueError(f"Only `years` is supported for {self.time_unit}")
 
     def tick(self, delta_years: FixedPoint) -> None:
         """ticks the time by delta_time amount"""
-        self.time += delta_years
+        self._time += delta_years
 
     def step(self) -> None:
         """ticks the time by step_size"""
-        self.time += self.step_size
+        self._time += self.step_size
+
+    def time_conversion(self, unit: TimeUnit = "seconds") -> FixedPoint:
+        """Convert time to different units
+
+        .. todo:: we will need to add conditions for self.time_unit in each conversion type
+        """
+        if unit == "seconds":  # 31,556,952 seconds in a year
```

Comment:
storing TimeUnit as an Enum with the relevant scalar as the number would eliminate this if else block

Reply:
Ok, I did this. But really though we will want to redo this with stdlib time utilities soon. I just wanted to fix the errors with the setters.